src/controllers/attachment/index.ts

```diff
@@ -91,28 +124,27 @@ export class attachment extends Controller {
   @Get('/{id}')
   @Response<NotFound>(404)
   @Response<BadRequest>(400)
+  @Produces('application/octet-stream')
+  @Produces('application/json')
   @SuccessResponse(200)
-  public async getById(
-    @Request() req: express.Request,
-    @Path() id: string,
-    @Header('return-type') type: 'json' | 'file'
-  ): Promise<unknown | Readable> {
+  public async getById(@Request() req: express.Request, @Path() id: string): Promise<unknown | Readable> {
     this.log.debug(`attempting to retrieve ${id} attachment`)
-    const { accept } = req.headers
     const [attachment] = await this.db.attachment().where({ id })
     if (!attachment) throw new NotFound('attachment')
     const { filename, binary_blob } = attachment
 
-    // log and default to octect-stream
-    if (type === 'json' || accept === 'application/json') {
-      this.setHeader('content-type', 'application/json')
-      try {
-        return JSON.parse(binary_blob)
-      } catch (err) {
-        this.log.warn('requested type failed, returning as octet')
+    const orderedAccept = parseAccept(req.headers.accept || '*/*')
+    if (filename === 'json') {
+      for (const mimeType of orderedAccept) {
+        if (mimeType === 'application/json' || mimeType === 'application/*' || mimeType === '*/*') {
+          const json = JSON.parse(binary_blob)
+          return json
+        }
+        if (mimeType === 'application/octet-stream') {
```

Comment:
like this check looks useless since we are going to return `return this.octetResponse(binary_blob, filename)` response

Reply:
We're in a loop so no it's not useless. If some one asks for a random mimetype (not covered here) and as a second preference an `application/json` response you would not want it to return an octet stream